torch/_functorch/config.py

```diff
@@ -64,6 +64,11 @@
 # cost of some performance
 aggressive_recomputation = False
 
+# Unlifts effect tokens from the inputs/outputs in the traced graph and instead
+# inserts make_token/sink_token calls in the graph to create tokens and then
+# sink them at the end.
+unlift_effect_tokens = False
```

Comment:
Hmm, I do wonder if we should just set the default here to True - given that Elias said handling the source/sink ops in the graph is easier than having to worry about the extra graph args.

The downside is obviously that non-inductor backends that opt into AOTAutograd need to worry about handling these extra ops in the graph. Although they'll only see these ops in the first place if there are with-effect-related ops in the user's program

Idk if @eellison / @zou3519 / @ezyang have any thoughts. Maybe the answer is that should have real docs for "here is what you should expect in the graph if you are a compiler below AOTAutograd", and that would make a decision like this easier.

Reply:
If the contract is that AOTAutograd gives a functional graph, then the safe default should be False. Setting it to True means that we get a graph that isn't functional, which will likely lead to silent errors with the backend when these tokens are present unless the backend knows how to handle it.

Agreed on the docs, we should leave some docs. I imagine that most backends will just set this to True, but this should be a conscious choice -- if the backend sets this to True, then they need to understand that the special source/sink nodes exist and aren't functional.